elements/lisk-p2p/src/peer_directory/base_list.ts

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright © 2019 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+import { P2PPeerInfo } from '../p2p_types';
+import { constructPeerIdFromPeerInfo, getBucket, PEER_TYPE } from '../utils';
+
+export interface PeerListConfig {
+	readonly peerBucketCount: number;
+	readonly peerBucketSize: number;
+	readonly secret: number;
+	readonly peerType: PEER_TYPE;
+}
+export interface CustomPeerInfo {
```

Comment:
I'd extend from the regular peer info instead of creating a new one. In that case, we should move this to `p2p_types`.

Reply:
Actually this type is confined to peer_directory and we never want to disclose `CustomPeerInfo` to upper classes so this is the reason to keep it here. 
And we are not extending it from peerInfo but we are keeping it as a field and this is more like wrapper around it. Its because if a custom field is overwritten by peerInfo coming from the network it can manipulate or affect the peer_bucket functionality